util/types/json/impl.go

```diff
@@ -57,8 +64,9 @@ func normalize(in interface{}) JSON {
 			array[i] = normalize(elem)
 		}
 		return jsonArray(array)
+	default:
+		panic("unsupported json type")
```

Comment:
after serialize, there is no jsonDecimal but jsonDouble instead.
you can see in MySQL 5.7 source code, Decimal, Integer and Double have same compare-martix.